plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py

```diff
@@ -200,18 +214,58 @@ def sweep(self, arguments: List[str]) -> None:
             returns = self.launcher.launch(overrides, initial_job_idx=self.job_idx)
             self.job_idx += len(returns)
             for trial, ret in zip(trials, returns):
-                study._tell(trial, optuna.trial.TrialState.COMPLETE, [ret.return_value])
+                values: Optional[List[float]] = None
+                state: optuna.trial.TrialState = optuna.trial.TrialState.COMPLETE
+                func_error: Optional[Exception] = None
+                try:
+                    if len(directions) == 1:
+                        values = [float(ret.return_value)]
```

Comment:
can you raise a proper error if the user does not return a floating point value here?
blindly casting to float is not great.

Reply:
Thank you for pointing it out. I raised `ValueError` here in https://github.com/facebookresearch/hydra/pull/1294/commits/30b50e48e4a0d073ea7c892b4cddae8808ff33e6.